src/host_abi.zig

```diff
@@ -0,0 +1,106 @@
+/// All Roc functions that are exposed to the host take 1 argument and return void.
+/// The 1 argument is a pointer to one of these structs, which includes an address
+/// that the Roc call will write the return value into. This design makes Roc's ABI
+/// very simple; the calling convention is just "pass a single pointer".
+pub const RocCall = struct {
+    /// Function pointers that the Roc program uses, e.g. alloc, dealloc, etc.
+    ops: *RocOps,
```

Comment:
This `ops` pointer is what will actually get threaded through from one Roc function call to another behind the scenes.

`arg` and `ret` only get used in the outermost function wrapper that the host calls.